internal/csi-addons/rbd/replication.go

```diff
@@ -886,11 +886,14 @@ func (rs *ReplicationServer) GetVolumeReplicationInfo(ctx context.Context,
 // RemoteStatus returns one SiteMirrorImageStatus item from the SiteStatuses
 // slice that corresponds to the remote site's status. If the remote status
 // is not found than the error ErrNotExist will be returned.
-func RemoteStatus(gmis *librbd.GlobalMirrorImageStatus) (librbd.SiteMirrorImageStatus, error) {
+func RemoteStatus(ctx context.Context, gmis *librbd.GlobalMirrorImageStatus) (librbd.SiteMirrorImageStatus, error) {
 	var (
 		ss  librbd.SiteMirrorImageStatus
 		err error = librbd.ErrNotExist
 	)
+	log.DebugLog(
+		ctx,
+		"Site Statuses: %s", gmis.SiteStatuses)
```

Comment:
this logging includes `%!s` in the output, it seems `gmis.SiteStatuses` can not simply be formatted as a string, does the struct have a `String()` function?

Reply:
No it doesn't have it.

Reply:
If you want to log it, you will either need to log attributes of if, or write a `String()` function for it.

Reply:
can you do the logging at line `898` inside for loop of `for i := range gmis.SiteStatuses {` but before `if` check? so that we don't need to log multiple times?